WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/FetchVerticalsUseCase.kt

```diff
@@ -0,0 +1,51 @@
+package org.wordpress.android.ui.sitecreation.usecases
+
+import kotlinx.coroutines.experimental.suspendCancellableCoroutine
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.VerticalActionBuilder
+import org.wordpress.android.fluxc.store.VerticalStore
+import org.wordpress.android.fluxc.store.VerticalStore.FetchVerticalsError
+import org.wordpress.android.fluxc.store.VerticalStore.FetchVerticalsPayload
+import org.wordpress.android.fluxc.store.VerticalStore.OnVerticalsFetched
+import org.wordpress.android.fluxc.store.VerticalStore.VerticalErrorType.GENERIC_ERROR
+import javax.inject.Inject
+import kotlin.coroutines.experimental.Continuation
+
+/**
+ * Transforms EventBus event to a coroutines.
+ *
+ * The client may dispatch multiple requests, but we want to accept only the latest one and ignore all others.
```

Comment:
The first thing I have done when I looked at this class was to read this comment and I am afraid I didn't fully understand it. If we want to have a comment like this here, it might help to provide a bit more context about what it's referring to.

Especially the `job.cancel()` part is somewhat confusing to me.